src/main/java/hudson/maven/reporters/TestMojo.java

```diff
@@ -231,12 +245,41 @@ public static TestMojo lookup(String artifactId, String groupId, String goal) {
     
     public static TestMojo lookup(MojoInfo mojo) {
         TestMojo testMojo = lookup(mojo.pluginName.groupId, mojo.pluginName.artifactId, mojo.getGoal());
+
+        if (testMojo == null && getReportsDirectoryOverride(mojo) != null) {
+			testMojo = FALLBACK;
+        }
+
         if (testMojo != null && testMojo.canRunTests(mojo)) {
             return testMojo;
         }
+
         return null;
     }
-    
+
+    /**
+     * For an unknown test-capable MOJO, returns the path to its reports directory as configured by a Maven property.
+     * @param mojo An unknown MOJO.
+     * @return the value of the expression <code>${jenkins.&lt;mojo-execution-id&gt;.reportsDirectory}</code>.
+     */
+    private static String getReportsDirectoryOverride(MojoInfo mojo) {
+        // Validity of the override property should be constrained to "test" and "integration-test" phases but there seems to be no
+        // way to check this, as MojoExecution.getLifecyclePhase() (added 2009-05-12) causes tests to throw NoSuchMethodError!!!
+        // So, we're obliged to assume that the property is configured for a valid test life cycle phase.
+//        String phase = mojo.mojoExecution.getLifecyclePhase();
+//        if ("test".equals(phase) || "integration-test".equals(phase)) {
+        try {
```

Comment:
But what would the TODO action be? The code comment is attempting to explain what it would prefer to do at this point and the reason why it cannot do so. It's unclear to me why the commented code compiles but won't execute. If I remove this block of code altogether we've lost track of the important point that we ought really to constrain executions to testing phases.